### PR TITLE
[BUILD] Use fully qualified references to trace/common namespace

### LINKED ISSUE #2417: 'SpanContext' is not a member of 'opentelemetry::v1::sdk::trace'
Hello! I am trying to add the OpenTelemetry SDK as a dependency into our project, and I am consistently hitting this compile issue. I would appreciate some help.

**Describe your environment** Describe any aspect of your environment relevant to the problem, including your platform, build system, version numbers of installed dependencies, etc. If you're reporting a problem with a specific version of a library in this repo, please check whether the problem has been fixed on main branch.

* CentOS in Parallel 18
* The latest release `opentelemetry-cpp-1.12.0.tar.gz` downloaded

**Steps to reproduce**
Describe exactly how to reproduce the error. Include a code sample if applicable.

 When I include certain dependencies from metrics SDK, for example any of the following
```
#include <opentelemetry/sdk/metrics/export/periodic_exporting_metric_reader_factory.h>
#include <opentelemetry/sdk/metrics/meter.h>
#include <opentelemetry/sdk/metrics/meter_provider.h>
#include <opentelemetry/sdk/metrics/meter_provider_factory.h>
```

and compile with either Bazel or CMake, I see the error messages pasted below

My Makefile:
```
build: download-src
	${MKDIR_P} build
	cd build && \
	${CMAKE} -DCMAKE_INSTALL_PREFIX=../installed \
		-DCMAKE_CXX_STANDARD=17 \
		-DWITH_STL=CXX17 \
		-DWITH_ABSEIL=ON \
		-Dabsl_ROOT=../../abseil/installed \
		-DBUILD_TESTING=OFF \
		../src
```

Bazel targets includes the following targets in the dependency
```
        "@io_opentelemetry_cpp//api",
        "@io_opentelemetry_cpp//exporters/ostream:ostream_metric_exporter",
        "@io_opentelemetry_cpp//exporters/ostream:ostream_span_exporter",
        "@io_opentelemetry_cpp//sdk:headers",
        "@io_opentelemetry_cpp//sdk/src/metrics",
        "@io_opentelemetry_cpp//sdk/src/trace",
        "@opentelemetry-protos//:otel_exporter_protos",
```

**What is the expected behavior?**
That it compiles.

**What is the actual behavior?**
What did you see instead?

```
bazel-out/aarch64-fastbuild/bin/external/io_opentelemetry_cpp/sdk/_virtual_includes/headers/opentelemetry/sdk/metrics/data/exemplar_data.h:29:53: error: 'SpanContext' is not a member of 'opentelemetry::v1::sdk::trace' 29 | static ExemplarData Create(std::shared_ptr<trace::SpanContext> context, | ^~~~~~~~~~~ bazel-out/aarch64-fastbuild/bin/external/io_opentelemetry_cpp/sdk/_virtual_includes/headers/opentelemetry/sdk/metrics/data/exemplar_data.h:29:64: error: template argument 1 is invalid 29 | static ExemplarData Create(std::shared_ptr<trace::SpanContext> context, | ^ bazel-out/aarch64-fastbuild/bin/external/io_opentelemetry_cpp/sdk/_virtual_includes/headers/opentelemetry/sdk/metrics/data/exemplar_data.h:50:16: error: 'SpanContext' in namespace 'opentelemetry::v1::sdk::trace' does not name a type 50 | const trace::SpanContext &GetSpanContext() const noexcept { return context_; } | ^~~~~~~~~~~
```

**Additional context**

Compiling with these header files would work properly

```
#include <opentelemetry/exporters/ostream/metric_exporter_factory.h>
#include <opentelemetry/exporters/ostream/span_exporter_factory.h>
#include <opentelemetry/metrics/provider.h>
#include <opentelemetry/sdk/metrics/aggregation/default_aggregation.h>
#include <opentelemetry/sdk/metrics/aggregation/histogram_aggregation.h>
#include <opentelemetry/sdk/metrics/push_metric_exporter.h>
#include <opentelemetry/sdk/metrics/view/instrument_selector_factory.h>
#include <opentelemetry/sdk/metrics/view/meter_selector_factory.h>
#include <opentelemetry/sdk/metrics/view/view_factory.h>
#include <opentelemetry/sdk/trace/exporter.h>
#include <opentelemetry/sdk/trace/processor.h>
#include <opentelemetry/sdk/trace/simple_processor_factory.h>
#include <opentelemetry/sdk/trace/tracer_provider_factory.h>
#include <opentelemetry/trace/provider.h>
#include <opentelemetry/trace/span_context.h>
```

I can even compile with a declaration of SpanContext
```
   std::shared_ptr<opentelemetry::v1::trace::SpanContext> context;
```

### PR DESCRIPTION
**Pull request by bogdandrutu**
Fixes #2417

### CODE REVIEW

File: sdk/include/opentelemetry/sdk/metrics/exemplar/reservoir.h [hunk 1]
```diff
@@ -72,7 +72,7 @@ class ExemplarReservoir
       size_t size,
       std::shared_ptr<ReservoirCellSelector> reservoir_cell_selector,
       std::shared_ptr<ExemplarData> (ReservoirCell::*map_and_reset_cell)(
-          const common::OrderedAttributeMap &attributes));
+          const opentelemetry::common::OrderedAttributeMap &attributes));
```
Comment (ThomsonTan) [inline sdk/include/opentelemetry/sdk/metrics/exemplar/reservoir.h:75]: This seems under SDK, not API namespace.

Comment (bogdandrutu) [reply sdk/include/opentelemetry/sdk/metrics/exemplar/reservoir.h]: Done.

**Comment by bogdandrutu** [comment]
@ThomsonTan unfortunately on my mac format produces crazy results (maybe some hints on how to properly run would be appreciated), is it possible to help me with formatting this code?

**Comment by lalitb** [comment]
> @ThomsonTan unfortunately on my mac format produces crazy results (maybe some hints on how to properly run would be appreciated), is it possible to help me with formatting this code?

@bogdandrutu  - You can do this by
- installing and running clang-format-10. This is the default package in ubuntu 20.04, so you can create a docker container to do that ( refer - https://github.com/open-telemetry/opentelemetry-cpp/blob/v1.12.0/ci/install_format_tools.sh)
 - copy and apply the patch given by the format failure in the build logs.

**Comment by esigo** [comment]
@bogdandrutu you could use docker:

```shell
./ci/run_docker.sh  tools/format.sh
```

**Review by marcalff — APPROVED** [review]
LGTM, thanks for the fix.

**Review by owent — APPROVED** [review]
LGTM after the format been fixed.

**Comment by marcalff** [comment]
@bogdandrutu 

> @ThomsonTan unfortunately on my mac format produces crazy results (maybe some hints on how to properly run would be appreciated), is it possible to help me with formatting this code?

Applied clang-format-10 and pushed to your branch.

**Pull request merged**